index.bs

```diff
@@ -12269,7 +12269,16 @@ to Consider</a>
 
 	In practice however, this merely allows deduction of information
 	already readily available by easier means (User Agent string),
-	such as "this is Chrome running on an x86".
+	such as "this is Chrome running on an x86".  However, to reduce the
+	possibility of additional fingerprinting, we mandate browsers take
+	action to mitigate fingerprinting issues.  The exact approach is not
+	prescribed, but one possible approach is described in the paper by
+	Pierre Laperdrix, Benoit Baudry, Vikas Mishra. <a
+	href="https://hal.inria.fr/hal-01527580/document">"FPRandom: Randomizing
+	core browser objects to break advanced device fingerprinting
+	techniques"</a>, ESSoS 2017 - 9th International Symposium on Engineering
+	Secure Software and Systems, Jul 2017, Bonn,
+	Germany. pp.17. hal-01527580.
```

Comment:
I'll note that this approach, and the variant Brave ships, are both trivially defeated (this is not theoretical). I'd feel more comfortable not linking to this, because it doesn't work. The spec should not promote techniques that don't work.

Reply:
Removing the link is acceptable to me.

Reply:
@pes10k any comment?  @padenot, any pointers to work re: defeating farbling?

Reply:
I have no opinion about whether there is a link in the text. I especially do not want to give any impression of pushing for Brave's approach to be in the standard (i don't think think it should be).

I do think that "trivially defeated" (whether we agree or not in this particular case) are unrated on the web, since 1) making something "difficult but doable" is a deterrent in many cases, 2) making it less pleasant to do perf-wise really does deter some 1ps from doing it, 3) (prob most important) it helps put a bigger gap between benign and malicious API uses, which allows for other web-compat interventions.

Reply:
I note that if the referenced paper is removed, then the sentence in lines 12273-12274 basically implies all the other things that follow including handling the sample rate in PR #2223.  So I think we should either not do #2223 or not do this. Or refine the sentence to require a mitigation if a narrower scope.   (Then I'm not sure what to say here, though.)

Reply:
As [previously discussed](https://github.com/w3cping/tracking-issues/issues/53#issuecomment-653541639), the fingerprinting offered by the Web Audio API and that we see in use today is just math. Essentially what is being fingerprinted is the implementation of the DSP algorithm of browsers and their environment, and not a particular the user or unique instance of a browser. 

What would be rational however is to encourage implementers to be careful at what math function are being used in a Web Audio API implementation and where they are from to avoid exposing the math library of the OS, that can (and does) change. This kind of prevent fingerprinting via the math library, which is kind of the same as finding the OS and its version, but some OSes can update their `libm` without a major version change, and so is probably not terribly useful.

This would only be useful if both:
- The math function that comes from the OS can be used to fingerprint
- The input of this function is controllable enough from the Web Audio API, where the values are restricted rather often, if controllable at all

In practice: Firefox (I tested different versions) on different computers with macOS 10.14 and macOS 10.15 and Chrome (I also tested different version) on different computers with the same OS pair return the same hash each time, regardless of the version of the "Web Audio Fingerprinting" I use, showing that this hash is function of the browser version.

More data point would be useful.

Reply:
@padenot i linked to some existing papers / data on web-audio fingerprinting here: https://github.com/WebAudio/web-audio-api/issues/2061#issuecomment-632372677

Specifically, 
> The most recent panopticlick data sees ~44 different web audio fingerprints; the Online Tracking: A 1-million-site Measurement and Analysis sees 713 different audio fingerprints during just the single study period

I dont know if thats compatible with the the "its just libm / implementation" conclusion (im not arguing it isn't right now) but is that the kind of data you're looking for?

Reply:
This is known, yes. The kind of data I'm looking for is the opposite: get a bunch of machines in a controlled environment and check what makes the fingerprint differ. In my own testing, it's the browser, its version and the OS, and sometimes its version.

Reply:
Also yes it's very much compatible at first glance, but it would be useful to have a repartition.